### PR TITLE
[OBSDEF-7964] ObjectScale Gateway: add fluxd endpoint

### PR DESCRIPTION
**Pull request by victor-polyansky**
## Purpose
[OBSDEF-7964](https://jira.cec.lab.emc.com:8443/browse/OBSDEF-7964)
Added nginx config for fluxd endpoint with auth through fedsvc.

Queries are handled for ObjectScale (`/fluxd` endpoint) and objectstores (`/fluxd/<objstore_namespace>/<objstore_name>` endpoint).
```
$ curl -ik https://<gateway_ip>/mgmt/login -u emcmonitor:ChangeMe
$ curl -ik -XPOST -H 'accept:application/csv' -H 'content-type:application/vnd.flux' -H "X-SDS-Auth-Token: $TOKEN" https://<gateway_ip>/flux/ -d @query.txt
$ curl -ik -XPOST -H 'accept:application/csv' -H 'content-type:application/vnd.flux' -H "X-SDS-Auth-Token: $TOKEN" https://<gateway_ip>/flux/<objstore_namespace>/<objstore_name>/ -d @query.txt
```

Required changes in objectscale-gateway image: https://eos2git.cec.lab.emc.com/ECS/federation-service/pull/70

## PR checklist
- [ ] make test passed
- [x] make build passed
- [x] helm install <chart> passed
- [x] vSphere7 make package deployments passed
- [ ] helm upgrade <chart> passed
- [ ] helm test <release_name> passed
- [ ] Passed - https://asd-ecs-jenkins.isus.emc.com/jenkins/view/ECS-Flex/job/charts-custom-ci/

## Testing
https://asd-ecs-jenkins.isus.emc.com/jenkins/job/charts-custom-ci/183/

Deployed on beige-b in `dellemc-po-ly-av-007-domain-c79` namespace with objectscale-gateway custom image.

### CODE REVIEW

**Review by vazhem — APPROVED** [review]

**Comment by SeemaTahaliyani** [comment]
@victor-polyansky can you do quick test if not alreday that object-gateway pod comes up even if --set global.monitoring.enabled=false which is what we do normally on minikube. Thanks.

**Comment by victor-polyansky** [comment]
> @victor-polyansky can you do quick test if not alreday that object-gateway pod comes up even if --set global.monitoring.enabled=false which is what we do normally on minikube. Thanks.

@SeemaTahaliyani Checked and found that it doesn't come up. Added handling of this case in config, now pod comes up. Thanks.

File: objectscale-gateway/templates/configmap.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+{{- include "objectscaleGateway.labels" . | nindent 4 }}
+data:
+{{- if .Values.global.monitoring.enabled }}
+  location_flux.conf: |
+    # handle ObjectScale fluxd
+    location = /flux/ {
+{{- include "objectscaleGateway.dnsResolver" . | nindent 6 }}
+      auth_request /validate_token;
+      rewrite /flux/ /api/external/v3/query break;
+      proxy_pass http://{{ .Release.Name }}-fluxd.{{ .Release.Namespace }}.svc.cluster.local:8093;
```
Comment (SeemaTahaliyani) [inline objectscale-gateway/templates/configmap.yaml:17]: is it necessary to add .svc.cluster.local?
I think it should resolve by itself given the below resolve conf
`[root@10:~]$ k exec -it objectscale-gateway-7d766df5f7-zsnk5 bash
bash-4.4# cat /etc/resolv.conf
nameserver 10.96.0.10
search default.svc.cluster.local svc.cluster.local cluster.local lss.emc.com
options ndots:5
bash-4.4#
`

File: objectscale-gateway/templates/configmap.yaml [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+{{- include "objectscaleGateway.dnsResolver" . | nindent 6 }}
+      auth_request /validate_token;
+      rewrite /flux/([a-z0-9-]+)/([a-z0-9-]+)/ /api/external/v3/query break;
+      proxy_pass http://$2-fluxd.$1.svc.cluster.local:8093;
+    }
+
+    location /validate_token {
+      internal;
+      proxy_buffering            on;
+      proxy_pass http://fedsvc/mgmt/validateToken;
```
Comment (SeemaTahaliyani) [inline objectscale-gateway/templates/configmap.yaml:31]: Is fluxd and grafana going be always in the objectscale namepace?
If not then url should be **http://fedsvc.(objectScaleNamespace)/mgmt/validateToken**

**Comment by SeemaTahaliyani** [comment]
@victor-polyansky Will approve, but left couple of comments if you can answer and if any change is needed, i am fine if you separately too. Thanks.

**Review by SeemaTahaliyani — APPROVED** [review]

**Pull request merged**